test/extended/placements/placements.go

```diff
@@ -0,0 +1,113 @@
+		// Skip for non-supported test environments
+		exutil.PreCheckEnvSupport(k, "kcp-stable.apps.kcp-internal", "kcp-unstable.apps.kcp-internal")
+		myWs := k.WorkSpace()
+
+		g.By("# Validate apibinding creation is successful in user home workspace")
+		myAPIBinding := apb.NewAPIBinding(apb.SetAPIBindingReferencePath("root:redhat-acm-compute"), apb.SetAPIBindingReferenceExportName("kubernetes"))
+		myAPIBinding.Create(k.WithSpecificWorkSpaceServer(myWs))
+
+		g.By("# Verify that default placement has been created for shared compute")
+		defaultPlacementSc, err := k.WithoutNamespace().WithoutKubeconf().Run("get").Args("placement", "default", "-o", "json").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(defaultPlacementSc).Should(o.And(
+			o.ContainSubstring("locationWorkspace"),
+			o.ContainSubstring("root:redhat-acm-compute"),
+		))
+
+		g.By("# Verify that APIBinding will be annotated with workload.kcp.dev/skip-default-object-creation")
+		apiBindingAnnotation, err := k.WithoutNamespace().WithoutKubeconf().Run("get").Args("apibinding", myAPIBinding.Metadata.Name, "-o", "json").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(apiBindingAnnotation).Should(o.ContainSubstring("workload.kcp.dev/skip-default-object-creation"))
+
+		g.By("# Verify creating a namespace in the current workspace")
+		k.WithoutKubeconf().Run("create").Args("namespace", "test52823").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("# Verify that it randomly select a location and bind to all namespaces in this workspace")
+		namespaces := []string{"default", "test52823"}
+		for _, namespace := range namespaces {
+			out, err := k.WithoutNamespace().WithoutKubeconf().Run("get").Args("namespace", namespace, "-o", "json").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(out).Should(o.And(
+				o.ContainSubstring("scheduling.kcp.dev/placement"),
+				o.ContainSubstring("state.workload.kcp.dev/"),
+				o.ContainSubstring("Sync"),
+			))
+		}
+
+		g.By("# verify default placement has been created for BYO clusters")
+		pclusterKubeconfig := os.Getenv("PCLUSTER_KUBECONFIG")
+		if pclusterKubeconfig == "" {
+			g.Skip("No pcluster kubeconfig set for the test scenario")
+		}
+
+		k.SetupWorkSpaceWithNamespace()
+		myWs = k.WorkSpace()
+		k.SetPClusterKubeconf(pclusterKubeconfig)
+
+		g.By("# Create workload sync and generate syncer resources manifests in current workspace")
+		mySyncer := nsc.NewSyncTarget()
+		mySyncer.OutputFilePath = "/tmp/" + myWs.Name + "." + mySyncer.Name + ".yaml"
+		mySyncer.Create(k)
+		defer mySyncer.Clean(k)
+
+		g.By("# Apply syncer resources on pcluster and wait for synctarget become ready")
+		defer k.AsPClusterKubeconf().WithoutNamespace().WithoutWorkSpaceServer().Run("delete").Args("-f", mySyncer.OutputFilePath).Execute()
+		err = k.AsPClusterKubeconf().WithoutNamespace().WithoutWorkSpaceServer().Run("apply").Args("-f", mySyncer.OutputFilePath).Execute()
```

Comment:
When AsPClusterKubeconf() is used, it does not make sense to involve Workspace server in terms of calling WithoutWorkSpaceServer(). How about updating https://github.com/kcp-dev/kcp-tests/blob/main/test/extended/util/client.go `if !c.withoutWorkSpaceServer` to `if !c.asPClusterKubeconf && !c.withoutWorkSpaceServer` such that you need no call WithoutWorkSpaceServer()?

Reply:
Thanks, Done, Addressed !!